### ISSUE TITLE
[Feature Request] Folder Thumbnail Previews

### ISSUE THREAD

**Issue by P650SE** [hardcore-sushi/DroidFS#136]
First of all, thanks for all your efforts to develop this app.

I would like to request the addition of folder thumbnails. This feature would display small thumbnails inside each folder icon, showing a preview of the images, and other content inside the folder. It could be implemented as an optional toggle in the settings menu - to be enabled at the user's preference.

I created a few concepts (attached to this post) of how I imagine this could look. The advantage of these folder thumbnails would be significant in navigation - as it allows the user to preview the contents of each folder from outside, rather than clicking onto each one and backing out repeatedly in order to find the desired destination.

Hope this idea is of interest to you!

![Concept 1](https://user-images.githubusercontent.com/60034877/170742600-6f216e74-ca1b-4dce-9d93-07d572722121.png)
![Concept 2](https://user-images.githubusercontent.com/60034877/170742603-a1a83c28-85a5-46cf-a382-eb5065f64412.png)
![Concept 3](https://user-images.githubusercontent.com/60034877/170742605-89e2cd3b-65a2-4283-9980-cf6479a7966a.png)

**Comment by hardcore-sushi** [comment]
Yes that could be a good idea. But how are selected the pictures displayed in the folder thumbnail? The first according to the sort order, or randomly? And should a folder also show images from subdirectories?

**Comment by P650SE** [comment]
Glad for your reply! I suggest by default that the selected pictures should be sorted according to alphabetical (or numerical) order, as the first pictures in a folder are typically the most recognisable and easily relatable. Another idea is that the selected image thumbnails could be determined by the sort order of contents inside the folder. So if the order of the folder contents is set to date, then the icon thumbnails are also sorted by most recent first. And likewise, if the order of the contents inside the folder is set to name, then the icon thumbnails are sorted alphabetically.

You could also have a global checkbox that allows to override the sort order for all folders at once, rather than needing to click onto each folder and configure it independently.

For the question of whether a folder should show images from subdirectories, if I understand correctly, I guess that most file explorers typically wouldn't do this (show thumbnails for contents that are not directly inside the root) but perhaps it's something that could be experimented with and enabled as a user-configured option?
